src/helpers.js

```diff
@@ -87,28 +87,36 @@ const getCourseSectionFromFeatureUrl = courseFeature => {
   } else return featureUrl
 }
 
-const getCourseCollectionObject = courseCollection => {
-  const feature = courseCollection["ocw_feature"]
-  const subfeature = courseCollection["ocw_subfeature"]
-  const speciality = courseCollection["ocw_speciality"]
-  const collection = {}
-  if (feature) {
-    collection["topic"] = feature
-  }
-  if (subfeature) {
-    collection["subtopic"] = subfeature
-  }
-  if (speciality) {
-    collection["speciality"] = speciality
-  }
-  return collection
-}
-
-const getCourseCollectionText = (courseCollection, separator) => {
-  const collection = getCourseCollectionObject(courseCollection)
-  return Object.keys(collection)
-    .map(property => collection[property])
-    .join(` ${separator} `)
+const getConsolidatedTopics = courseCollections => {
+  const topics = {}
+  courseCollections.forEach(courseCollection => {
+    if (!topics.hasOwnProperty(courseCollection["ocw_feature"])) {
+      topics[courseCollection["ocw_feature"]] = {}
+    }
+    if (courseCollection["ocw_subfeature"]) {
+      if (
+        !topics[courseCollection["ocw_feature"]].hasOwnProperty(
+          courseCollection["ocw_subfeature"]
+        )
+      ) {
+        topics[courseCollection["ocw_feature"]][
+          courseCollection["ocw_subfeature"]
+        ] = []
+      }
+      if (courseCollection["ocw_speciality"]) {
+        if (
+          !topics[courseCollection["ocw_feature"]][
+            courseCollection["ocw_subfeature"]
+          ].includes(courseCollection["ocw_speciality"])
+        ) {
+          topics[courseCollection["ocw_feature"]][
+            courseCollection["ocw_subfeature"]
+          ].push(courseCollection["ocw_speciality"])
+        }
+      }
+    }
+  })
+  return topics
 }
```

Comment:
```suggestion
const getConsolidatedTopics = courseCollections => {
  let topics = {}
  courseCollections.forEach(courseCollection => {
    const { ocw_feature, ocw_subfeature, ocw_speciality } = courseCollection

    const collectionTopic = {
      [ ocw_feature ]: {}
    }
    if (ocw_subfeature) {
      collectionTopic[ocw_feature][ocw_subfeature] = [ocw_speciality].filter(
        Boolean
      )
    }
    topics = _.merge(topics, collectionTopic)
  })
  return topics
}
```

just thought htis function was a little hard to read, one suggestion for a way it could be simplified (tested it locally and the tests pass anyhow)